frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/__tests__/object-with-type-property-wrapper.spec.ts

```diff
@@ -0,0 +1,252 @@
+        exclusiveFlavor: {
+          orange: { color: 'green' },
+        },
+      },
+    });
+  });
+
+  it('appends correct type data', () => {
+    orangeSmoothie.asResource().attributes.price = {
+      value: 2,
+      currency: 'CZK',
+    };
+    orangeSmoothie.appendTypeData({ color: 'green' });
+    expect(orangeSmoothie.getTypeData()).toEqual({ ...orangeTypeData, color: 'green' });
+
+    orangeSmoothie
+      .appendType(SmoothieType.BANANA, { color: 'yellow' })
+      .appendType(SmoothieType.BANANA, { peel: { thickness: '3' } })
+      .appendType(SmoothieType.BANANA, {});
+
+    expect(orangeSmoothie.asResource()).toEqual({
+      attributes: {
+        price: {
+          value: 2,
+          currency: 'CZK',
+        },
+        exclusiveFlavor: {
+          banana: { color: 'yellow', peel: { thickness: '3' } },
+        },
+      },
+    });
+  });
+
+  it('appends correct type data for empty', () => {
+    emptySmoothie.appendTypeData({ color: 'green' });
+    expect(emptySmoothie.asResource()).toEqual({});
+    emptySmoothie.appendType(SmoothieType.BANANA, { color: 'green' });
+    expect(emptySmoothie.getTypeData()).toEqual({ color: 'green' });
+  });
+
+  it('set type data sanitize works', () => {
+    orangeSmoothie.setTypeData({ color: 'green', peel: { thickness: '5' } });
+    expect(orangeSmoothie.asResource()).toEqual({
+      attributes: {
+        exclusiveFlavor: {
+          orange: { color: 'green' },
+        },
+      },
+    });
+    emptySmoothie.setType(SmoothieType.STRAWBERRY, { color: 'red', seeds: { count: 3 } });
+    expect(emptySmoothie.getTypeData()).toEqual({ color: 'red' });
+  });
+
+  it('append type data sanitize works', () => {
+    orangeSmoothie
+      .appendTypeData({ color: 'green', peel: { thickness: '5' } })
+      .appendType(SmoothieType.ORANGE, { peel: { thickness: '4' } });
+    expect(orangeSmoothie.asResource()).toEqual({
+      attributes: {
+        exclusiveFlavor: {
+          orange: { ...orangeTypeData, color: 'green' },
+        },
+      },
+    });
+  });
+
+  it('set type data sanitize off', () => {
+    orangeSmoothie.setTypeData({ color: 'green', peel: { thickness: '5' } }, false);
+    expect(orangeSmoothie.asResource()).toEqual({
+      attributes: {
+        exclusiveFlavor: {
+          orange: { color: 'green', peel: { thickness: '5' } },
+        },
+      },
+    });
+    emptySmoothie.setType(SmoothieType.STRAWBERRY, { color: 'red', seeds: { count: 3 } }, false);
+    expect(emptySmoothie.getTypeData()).toEqual({ color: 'red', seeds: { count: 3 } });
+  });
+
+  it('append type data sanitize off', () => {
+    orangeSmoothie.appendTypeData({ color: 'green', peel: { thickness: '5' } }, false);
+    expect(orangeSmoothie.asResource()).toEqual({
+      attributes: {
+        exclusiveFlavor: {
+          orange: { ...orangeTypeData, color: 'green', peel: { thickness: '5' } },
+        },
+      },
+    });
+  });
+
+  it('stores inconsistent data', () => {
```

Comment:
I don't understand this test, can you write more informative comments? thanks

Reply:
added a comment few lines bellow